base-containers/base/bin/_run_student_intern

```diff
@@ -79,17 +78,34 @@ print("Chdir")
 os.chdir(start_cmd["working_dir"])
 print("Popen")
 print(fds)
-p = subprocess.Popen(shlex.split(start_cmd["command"]), preexec_fn=setlimits, stdin=fds[0], stdout=fds[1], stderr=fds[2])
 
+set_limits = lambda: set_limits_user(user)  # To know if the command should be executed as root or worker
+if start_cmd["command"] is None or start_cmd["command"] == "":  # Avoid to ron an empty command
+    start_cmd["command"] = "echo 'info: student container started with no command set' "
+
+# Run the command
+p = subprocess.Popen(shlex.split(start_cmd["command"]), preexec_fn=set_limits, stdin=fds[0], stdout=fds[1], stderr=fds[2])
 # Handle the signals
 print("Signal thread")
 thread = threading.Thread(target=lambda: handle_signals(p, client), daemon=True)
 thread.start()
-
 # Wait until the subprocess exit
 print("Waiting for process to end")
 retval = p.wait()
 
-# Exit with the same return value than the process (it will be given to docker and given back to run_student)
-print("Exiting")
-exit(retval)
+if start_cmd["ssh"]:
+    # Start ssh server
+    ssh_user, password = start_ssh_server(user)
+    # Send ssh information
+    ssh_user_bin = User[ssh_user].value  # 0 for root, 1 for worker
```

Comment:
Why not using "root" and "worker" instead of 0/1?

Reply:
Indeed. Actually encoding the user as a binary value (0 or 1) was to allow a constant size message.
Now since the message size is sent seperately, I will simply use "worker" and "root" instead :)